### PR TITLE
test: extend filesystem tests

### PR DESCRIPTION
**Pull request by matejvasek**
# Changes

- :test_tube: Extend filesystem tests.

### CODE REVIEW

File: pkg/filesystem/filesystem_test.go [hunk 1]
```diff
@@ -149,6 +87,95 @@ func TestFileSystems(t *testing.T) {
...
+		case fs.ModeSymlink:
+			t, _ := os.Readlink(path)
+			bs = []byte(t)
+		}
+		path, err = filepath.Rel(root, path)
+		if err != nil {
+			return err
+		}
+		files = append(files, FileInfo{
+			Path:       filepath.ToSlash(path),
```
Comment (dsimansk) [inline pkg/filesystem/filesystem_test.go:168]: How does it work with Windows backslash path separator?

Comment (matejvasek) [reply pkg/filesystem/filesystem_test.go:168]: The Go's `fs` package uses slashes (like zip or tar does).

Comment (matejvasek) [reply pkg/filesystem/filesystem_test.go:168]: This line is here precisely so it works on Windows.

**Comment by matejvasek** [comment]
@lance @jrangelramos  PTAL.

**Review by zroubalik — APPROVED** [review]
/lgtm
/approve

**Pull request merged**